A15-promote-reflection.md

```diff
@@ -0,0 +1,57 @@
+Promote Reflection
+----
+* Author(s): Carl Mastrangelo (carl-mastrangelo)
+* Approver: a11r
+* Status: Draft
+* Implemented in:
+* Last updated: 2018-05-30
+* Discussion at: https://groups.google.com/forum/#!topic/grpc-io/AOgFISlAgxk
+
+## Abstract
+
+Promote the Reflection Service from v1alpha to v1
+
+## Background
+
+Reflection is a means by which a server can describe what messages it
+supports.  Both the protocol and the [description](
+https://github.com/grpc/grpc/blob/master/doc/server-reflection.md) of the
+reflection service have not changed for a long time.
+
+## Proposal
+
+It is proposed that the package name (and corresponding directory structure)
+for [reflection.proto](
+https://github.com/grpc/grpc/blob/v1.12.x/src/proto/grpc/reflection/v1alpha/reflection.proto)
+be changed from `grpc.reflection.v1alpha` to `grpc.reflection.v1`.  Java and Go
```

Comment:
What about C and/or the wrapped languages?

Reply:
This part is less clear to me.  There is a C# and a C++ implementation, but I couldn't find implementations for the other langs.  I can't tell if this is intentional or not, so I was only planning on looking at those ones.

Reply:
I think it is good to know whether reflection is enabled by default in all the languages (C++ servers require users to explicitly link in a library). For C++ we can leave the user to decide which library to link in (or both).

Reply:
Can we update the status for C++ as requested by other reviewers.

Reply:
@a11r I have audited usages of the C++ implementation and have updated the plan here.   This looks doable with minimal code change and service interruption.

PTAL